integration/fuse/src/test/java/alluxio/fuse/AlluxioFuseFileSystemTest.java

```diff
@@ -174,9 +175,18 @@ public void open() throws Exception {
     when(mFileSystem.getStatus(expectedPath)).thenReturn(status);
     mFileInfo.flags.set(O_RDONLY.intValue());
 
-    // actual test
-    mFuseFs.open("/foo/bar", mFileInfo);
+    // Use another thread to open file so that
+    // we could change the file status when opening it
+    Thread t = new Thread(() -> mFuseFs.open("/foo/bar", mFileInfo));
+    t.start();
+    Thread.sleep(1000);
+    // If the file exists but is not completed, we will wait for the file to complete
     verify(mFileSystem).exists(expectedPath);
+    verify(mFileSystem, atLeast(10)).getStatus(expectedPath);
+    verify(mFileSystem, never()).openFile(expectedPath);
+
+    fi.setCompleted(true);
+    t.join();
```

Comment:
Could you add this test separate from the existing open test, ideally we test one case in each unit test.